Chefs/Services/Sharing/ShareService.cs

```diff
@@ -0,0 +1,72 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using Windows.ApplicationModel.DataTransfer;
+using WinRT;
+using WinRT.Interop;
+
+namespace Chefs.Services.Sharing;
+
+public class ShareService() : IShareService
+{
+	private Recipe? _recipe;
+	private IImmutableList<Step>? _steps;
+
+#if WINDOWS
+	private static readonly Guid _dtm_iid = new Guid("a5caee9b-8708-49d1-8d36-67d25a8da00c");
+	static IDataTransferManagerInterop DataTransferManagerInterop => DataTransferManager.As<IDataTransferManagerInterop>();
+#endif
+
+	public async Task ShareRecipe(Recipe recipe, IImmutableList<Step> steps, CancellationToken ct)
+	{
+		_recipe = recipe;
+		_steps = steps;
+
+#if WINDOWS
+		IntPtr result;
+		var hwnd = WindowNative.GetWindowHandle(App.MainWindow);
+		result = DataTransferManagerInterop.GetForWindow(hwnd, _dtm_iid);
+		DataTransferManager dataTransferManager = MarshalInterface<DataTransferManager>.FromAbi(result);
+		dataTransferManager.DataRequested += DataRequested;
+		DataTransferManagerInterop.ShowShareUIForWindow(hwnd, null);
+#else
+		var dataTransferManager = DataTransferManager.GetForCurrentView();
+        dataTransferManager.DataRequested += DataRequested;
+        DataTransferManager.ShowShareUI();
```

Comment:
change whitespace to tabs :)

![image](https://github.com/user-attachments/assets/36ad9d2c-0fb7-4a12-bcbc-b351d893da7b)